src/components/layer-controller/RasterChannelController.js

```diff
@@ -106,56 +106,56 @@ function RasterChannelController({
   handleChannelRemove,
   handleIQRUpdate,
   selectionIndex,
-  disableOptions = false,
+  isLoading,
+  use3d: newUse3d,
 }) {
   const { dtype } = getSourceFromLoader(loader);
   const [domain, setDomain] = useState(null);
   const [domainType, setDomainType] = useState(null);
+  const [use3d, setUse3d] = useState(null);
   const [selection, setSelection] = useState([{ ...channels[channelId].selection }]);
   const rgbColor = toRgbUIString(colormapOn, color, theme);
 
   useEffect(() => {
-    let mounted = true;
     if (dtype && loader && channels) {
-      const loaderSelection = [
-        { ...channels[channelId].selection },
-      ];
+      const loaderSelection = [{ ...channels[channelId].selection }];
       let domains;
       const hasDomainChanged = newDomainType !== domainType;
+      const has3dChanged = use3d !== newUse3d;
       const hasSelectionChanged = !isEqual(loaderSelection, selection);
-      if (hasDomainChanged || hasSelectionChanged) {
+      if (hasDomainChanged || hasSelectionChanged || has3dChanged) {
+        setUse3d(newUse3d);
         if (newDomainType === 'Full') {
           domains = [DOMAINS[dtype]];
           const [newDomain] = domains;
-          if (mounted) {
```

Comment:
I don't think we needed this `mounted` business unless I am missing something.

Reply:
I think we do, it must have been added because it is in a `useEffect`, and so the effect can potentially run after the component has been closed / cleaned up, causing errors / warnings in the console

Reply:
(Should only be needed in the `.then()` promise callback function)

Reply:
Can you remember why we needed it? Here's the PR it's from back when we were cleaning up all this view config business: https://github.com/vitessce/vitessce/pull/777  It seemed to be unmounting randomly here.

Reply:
We need it because if any of the `useEffect` dependencies change, then the effect function re-runs, and depending on the timing of the prior execution of the effect, the promise could resolve inside an un-mounted component. I think the `if(mounted)` check inside `useEffect` functions is a common pattern.

Reply:
OK this makes sense, especially with the frequency of renders.  The issue was setting the local 3d state outside the mount check.